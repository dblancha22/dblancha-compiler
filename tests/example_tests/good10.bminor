/* tests some more constructs in bminor */
z = -x; // unary negation
x^2; //exponantiotion
//simple test cases
x ^ 2
z != x % 2
z = 5 + 6;
z = 5 * 6;
z = 5 - 6;
z = 5 / 6;
q++;
z--;
z = 7 + 7 - z;

