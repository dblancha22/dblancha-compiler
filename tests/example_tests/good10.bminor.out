COMMENT /* tests some more constructs in bminor */
IDENTIFIER z
EQUALS =
MINUS -
IDENTIFIER x
SEMICOLON ;
COMMENT // unary negation
IDENTIFIER x
EXP ^
INTEGER_LITERAL 2
SEMICOLON ;
COMMENT //exponantiotion
COMMENT //simple test cases
IDENTIFIER x
EXP ^
INTEGER_LITERAL 2
IDENTIFIER z
NE !=
IDENTIFIER x
REMAINDER %
INTEGER_LITERAL 2
IDENTIFIER z
EQUALS =
INTEGER_LITERAL 5
PLUS +
INTEGER_LITERAL 6
SEMICOLON ;
IDENTIFIER z
EQUALS =
INTEGER_LITERAL 5
MULTIPLY *
INTEGER_LITERAL 6
SEMICOLON ;
IDENTIFIER z
EQUALS =
INTEGER_LITERAL 5
MINUS -
INTEGER_LITERAL 6
SEMICOLON ;
IDENTIFIER z
EQUALS =
INTEGER_LITERAL 5
DIVIDE /
INTEGER_LITERAL 6
SEMICOLON ;
IDENTIFIER q
INCREMENT ++
SEMICOLON ;
IDENTIFIER z
DECREMENT --
SEMICOLON ;
IDENTIFIER z
EQUALS =
INTEGER_LITERAL 7
PLUS +
INTEGER_LITERAL 7
MINUS -
IDENTIFIER z
SEMICOLON ;
